### ISSUE TITLE
Error loading SteamSdk.dll assembly

### ISSUE THREAD

**Issue by Mix-martes86** [KeenSoftwareHouse/SpaceEngineers#27]
Hi,

after several tedious hours just to download VS Community, I managed to successfully build the solution with no errors.
However, when launching, I get an Exception checking for Enum duplicities. Check it out:

```c#
static void TestEnumNotDuplicitiesInAssembly(Assembly assembly, HashSet<object> hashSet)
{
	foreach (Type type in assembly.GetTypes())

	MyEnumDuplicitiesTester.cs Line 76
	An unhandled exception of type 'System.Reflection.ReflectionTypeLoadException' occurred in mscorlib.dll
	Additional information: Unable to load one or several required types.
			
'assembly.ExportedTypes' threw an exception of type System.IO.FileNotFoundException
Unable to load file or assembly 'SteamSDK.dll' or one of its dependencies. Unable to find specified module.
```

File is not missing anywhere, so I don't get what's the problem.

Thanks in advance.

**Comment by sjsoft2015** [comment]
Just copy this lib from steam game folder... u can overwrite it 3rd folder.

**Comment by Mix-martes86** [comment]
That worked, and everything is just fine, though I keep getting these starting the game and loading a world, I'm guessing the fix for that is the same as the DLL:

```c#
DEFINITION_ERROR: , in file: C:\Users\Mix\Documents\Visual Studio 2013\Projects\SpaceEngineers-master\Sources\SpaceEngineers\bin\x86\Content\Data\Audio.sbc
Unable to find audio file: 'RealWepPlayRifleImpMetal', 'REAL\IMP\RealWepPlayRifleImpMetal3d01.wav'



   at Sandbox.Definitions.MyDefinitionErrors.Add(MyModContext context, String message, ErrorSeverity severity, Boolean writeToLog) in c:\Users\Mix\Documents\Visual Studio 2013\Projects\SpaceEngineers-master\Sources\Sandbox.Common\MyDefinitionErrors.cs:line 138

   at VRage.Audio.MyAudioExtensions.<.cctor>b__6(MySoundData cue, String message) in c:\Users\Mix\Documents\Visual Studio 2013\Projects\SpaceEngineers-master\Sources\Sandbox.Game\Game\Audio\MyAudioExtensions.cs:line 18

   at VRage.Audio.MyWaveBank.Add(MySoundData cue, MyAudioWave cueWave) in c:\Users\Mix\Documents\Visual Studio 2013\Projects\SpaceEngineers-master\Sources\VRage.Audio\MyWaveBank.cs:line 105

   at VRage.Audio.MyCueBank.InitWaveBank() in c:\Users\Mix\Documents\Visual Studio 2013\Projects\SpaceEngineers-master\Sources\VRage.Audio\MyCueBank.cs:line 160

   at VRage.Audio.MyCueBank..ctor(XAudio2 audioEngine, ListReader`1 cues) in c:\Users\Mix\Documents\Visual Studio 2013\Projects\SpaceEngineers-master\Sources\VRage.Audio\MyCueBank.cs:line 105

   at VRage.Audio.MyXAudio2.LoadData(MyAudioInitParams initParams, ListReader`1 sounds, ListReader`1 effects) in c:\Users\Mix\Documents\Visual Studio 2013\Projects\SpaceEngineers-master\Sources\VRage.Audio\MyXAudio2.cs:line 321

   at VRage.Audio.MyAudio.LoadData(MyAudioInitParams initParams, ListReader`1 sounds, ListReader`1 effects) in c:\Users\Mix\Documents\Visual Studio 2013\Projects\SpaceEngineers-master\Sources\VRage\Audio\MyAudio.cs:line 24

   at Sandbox.MySandboxGame.LoadData() in c:\Users\Mix\Documents\Visual Studio 2013\Projects\SpaceEngineers-master\Sources\Sandbox.Game\MySandboxGame.cs:line 996

   at Sandbox.MySandboxGame.Initialize() in c:\Users\Mix\Documents\Visual Studio 2013\Projects\SpaceEngineers-master\Sources\Sandbox.Game\MySandboxGame.cs:line 859

   at Sandbox.MySandboxGame.Run(Boolean customRenderLoop, Action disposeSplashScreen) in c:\Users\Mix\Documents\Visual Studio 2013\Projects\SpaceEngineers-master\Sources\Sandbox.Game\MySandboxGame.cs:line 290

   at SpaceEngineers.MyProgram.RunInternal(String[] args) in c:\Users\Mix\Documents\Visual Studio 2013\Projects\SpaceEngineers-master\Sources\SpaceEngineers\MyProgram.cs:line 239

   at SpaceEngineers.MyProgram.Main(String[] args) in c:\Users\Mix\Documents\Visual Studio 2013\Projects\SpaceEngineers-master\Sources\SpaceEngineers\MyProgram.cs:line 150

   at System.AppDomain._nExecuteAssembly(RuntimeAssembly assembly, String[] args)

   at System.AppDomain.ExecuteAssembly(String assemblyFile, Evidence assemblySecurity, String[] args)

   at Microsoft.VisualStudio.HostingProcess.HostProc.RunUsersAssembly()

   at System.Threading.ThreadHelper.ThreadStart_Context(Object state)

   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)

   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)

   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)

   at System.Threading.ThreadHelper.ThreadStart()
```

**Comment by kapitanov** [comment]
I've run into the same issue while trying to run a Debug build of the game, while a Release build went just fine. Looks like Debug and Release configurations use different versions of SteamSDK.dll - one is release and the other is debug.
Release version of SteamSDK.dll depends on VC++ Runtime libs - msvcp110.dll and msvcr110.dll, which are usually present in Windows folder and they are installed with Visual C++ Redistributable Packages.

However, debug version of SteamSDK.dll depends on debug versions of these libs (msvcp110d.dll and msvcr110d.dll) and they are installed with Visual Studio (not in any VS setup I guess).

Try to find these dlls and copy them into bin folder, that worked for me. Or install VS 2013 with all features selected - that should help too.

**Comment by ziron999** [comment]
to sjsoft I am having this exact same issue exactly where are these libs in the steam folder I don't see them anywhere?

**Comment by Evrey** [comment]
Cannot test SE, too. Gonna try copying the DLL from the game, when I'm back home.

The problem is, that the Source Code DLL (Last update from Nov 3, 2015.) seems to have wrong/broken dependencies. I asked DependencyWalker and it told me this:

* `msvcp120d.dll` & `msvcr120d.dll` missing
  My research suggests this to be rubbish, as the Visual Studio 2012 redists don't ship any debug DLLs. Debugged versions have to be linked statically via a compiler flag. Interestingly, re-installing VS2012 Update 4 fixed the problem for most people, although these are VS2013 DLLs. (I use Visual Studio 14, which is - as of now - the newest, and installing either redist of either VS version did not fix anything.)

* `api-ms-win-...` missing.
  There is a huge load of these guys missing. About a quarter of these DLLs are missing on Windows 7 and almost all of them are missing on Windows 10. Neither re-installing all VC redists nor installing all available windows SDKs fixed this. These seem to be Windows 8 API related.

* `ext-ms-win-...` missing on Windows 10.

* Some Internet Explorer related DLL has been missing on Windows 7. Why would I want to need that?

* `dcomp.dll` missing on Windows 7, which seems to be some Windows 8+ only stuff.

* ...

In a nutshell: I can compile SE now, without any warnings, but the new SteamAPI.dll is just broken, which prevents me from testing my stuff. If replacing the Source Code DLL will fix the problem for me, I'd suggest committing that one into the master branch.

**Comment by Evrey** [comment]
The old DLL fixed the problem for release builds.
